examples/multimodal/connect/descriptor.py

```diff
@@ -0,0 +1,247 @@
+# SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import asyncio
+import contextlib
+import json
+import logging
+import msgspec
+import nixl._api as nixl_api
+import nixl._bindings as nixl_bindings
+import numpy as np
+import os
+import signal
+import socket
+import torch
+import utils.nixl as nixl_utils
+import uuid
+
+from abc import ABC, abstractmethod
+from dynamo.sdk import async_on_start, depends, dynamo_context, dynamo_endpoint, service
+from dynamo.runtime import Backend, Client, Component, DistributedRuntime
+from enum import IntEnum
+from functools import cached_property
+from io import BytesIO
+from pydantic import BaseModel, ConfigDict, field_validator
+from pydantic_core import core_schema
+from typing import Any, Dict, List, Optional, Tuple
+
+from .connector import Connector
```

Comment:
this line I think is a circular dependency -

Reply:
🤔 but this class needs to know about `Connector`, how to allow both types to know about each other w/o causing a circular dependency?